File: problem-solving/baekjoon/11478_서로_다른_부분_문자열의_개수.go
// https://www.acmicpc.net/problem/11478
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var s string
	fmt.Fscanln(reader, &s)
	strs := map[string]bool{}
	for i := 0; i < len(s); i++ {
		for j := 1; i+j <= len(s); j++ {
			strs[s[i:i+j]] = true
		}
	}
	fmt.Fprintln(writer, len(strs))
}

File: problem-solving/baekjoon/11478_서로_다른_부분_문자열의_개수.py
# https://www.acmicpc.net/submit/11478
import sys

if __name__ == "__main__":
    s = sys.stdin.readline().rstrip()
    strs = {}
    for i in range(len(s)):
        for j in range(1, len(s)-i+1):
            strs[s[i:i+j]] = True
    print(len(strs))